File: AtlasWeb/Core/Models/LogisticMainViewModel.cs
using System;
using System.Collections.Generic;

namespace Core.Models
{
    public class LogisticWorkListViewModel
    {
        public DateTime StartDate { get; set; }
        public DateTime EndDate { get; set; }
        public string StockActionId { get; set; }
        public string TIFId { get; set; }
        public Int32 WorkListCount { get; set; }
        public string StateType { get; set; }
        public List<StockActionWorkListItemModel> stockactionlist { get; set; }
        public List<TTObjectClasses.MenuDefinition> menulist { get; set; }
        public List<WorkListItem> workListItems { get; set; }
        public List<WorkListItem> selectedWorkListItems { get; set; }
        public bool PartialCancel { get; set; }
        public bool EHUWait { get; set; }
        public bool IsEmergencyMaterial { get; set; }
    }

    public class StockActionWorkListItemModel
    {
        public string ObjectID { get; set; }
        public Int32 StockActionID { get; set; }
        public string Store { get; set; }
        public string DestinationStore { get; set; }
        public string StockactionName { get; set; }
        public string TransactionDate { get; set; }
        public string StateName { get; set; }
        public string StateFormName { get; set; }
        public string PatientName { get; set; }
        public WorkListResultColorEnum colorEnum { get; set; }
        public bool IsEmergencyMaterial { get; set; }
    }

    public class WorkListItem
    {
        public string ObjectDefName { get; set; }
        public string ObjectDefId { get; set; }
    }
    public enum WorkListResultColorEnum
    {
        red = 0,
        white = 1
    }
}

File: AtlasWeb/Core/wwwroot/app/Logistic/Models/LogisticMainViewModel.ts
import { BaseModel } from 'Fw/Models/BaseModel';
import { MenuDefinition } from 'NebulaClient/Model/AtlasClientModel';
import { TTListDef } from 'NebulaClient/StorageManager/DefinitionManagement/TTListDef';
import { Type } from 'NebulaClient/ClassTransformer';
import { Guid } from 'NebulaClient/Mscorlib/Guid';

export class LogisticMainViewModel extends BaseModel {

    stockactionlist: Array<StockActionWorkListItemModel> = new Array<StockActionWorkListItemModel>();
    @Type(() => Date)
    StartDate: Date;
    @Type(() => Date)
    EndDate: Date;
    StockActionId: string;
    TIFId: string;
    WorkListCount: number;
    StateType: string;
    menulist: Array<MenuDefinition>;
    workListItems: Array<WorkListItem>;
    selectedWorkListItems: Array<WorkListItem>;
    PartialCancel: boolean;
    EHUWait: boolean;
    IsEmergencyMaterial: boolean;
}

export class StockActionWorkListItemModel {

    public ObjectID: string;
    public StockActionID: number;
    public Store: string;
    public DestinationStore: string;
    public StockactionName: string;
    @Type(() => Date)
    public TransactionDate: Date;
    public StateName: string;
    public StateFormName: string;
    public PatientName: string;
}

export class QueryInputDVO {
    @Type(() => Guid)
    public StoreID: Guid;
    @Type(() => Date)
    public StartDate: Date;
    @Type(() => Date)
    public EndDate: Date;
    public StateType: string;
    public StockActionId: string;
    public TIFId: string;
    public WorkListCount: number;
    public SelectedWorkListItems: Array<WorkListItem>;
    public MKYSState: string;
    public PartialCancel: boolean;
    public EHUWait: boolean;
    public IsEmergencyMaterial: boolean;
}

export class MenuOutputDVO {
    public StockActionMenuItems: Array<MenuDefinition>;
    public WorkListSearchItem: Array<WorkListItem>;
    public StockDefinitionMenuItems: Array<TTListDef>;
}

export class WorkListItem {
    public ObjectDefName: string;
    public ObjectDefId: string;
}

export class DynamicComponentInfoDVO {
    public ComponentName: string;
    public ModuleName: string;
    public ModulePath: string;
    public objectID: string;
}

export enum WorkListResultColorEnum {
    red = 0,
    white = 1
}

export class StockMenuItem {
    public Name: string;
    public Value: number;
}

